### PR TITLE
Support dash topo when start the cluster

### PR DESCRIPTION
**Pull request by mapleFU**
Support inject topology for prometheus, grafana and alert manager.

### CODE REVIEW

**Comment by july2993** [comment]
you can merge master to fix the  ci

**Comment by mapleFU** [comment]
I've tested this pr on IDC:
```
ETCDCTL_API=3 etcdctl --endpoints="172.16.5.37:12375" get /topology --prefix=true

/topology/alertmanager
{"ip":"172.16.5.37","port":21332,"binary_path":"/home/pingcap/mapleFU/deploy/alertmanager-9093"}
/topology/grafana
{"ip":"172.16.5.37","port":30102,"binary_path":"/home/pingcap/mapleFU/deploy/grafana-3000"}
/topology/prometheus
{"ip":"172.16.5.37","port":9231,"binary_path":"/home/pingcap/mapleFU/deploy/prometheus-8249"}
```

I think you can review this pr now. @lonng @july2993 @breeswish

**Comment by lonng** [comment]
@mapleFU I think the binary path is not correct.

**Comment by mapleFU** [comment]
@lonng It's `deploy_dir` now, I think it's just the name we required.

File: pkg/operation/action.go [hunk 1]
```diff
@@ -59,9 +72,53 @@ func Start(
...
+		}
+	}
+
 	return nil
 }
 
+type topologyInfo struct {
+	IP        string `json:"ip"`
+	Port      int    `json:"port"`
+	DeployDir string `json:"deploy_dir"`
```
Comment (breeswish) [inline pkg/operation/action.go:105]: Looks like you need to change dashboard to use deploy_dir first?

**Comment by mapleFU** [comment]
I'll close this pr and reimplement it in another pull request.

**Pull request closed without merging**